sunbeam-python/sunbeam/plugins/validation/plugin.py

```diff
@@ -0,0 +1,139 @@
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+import click
+from packaging.version import Version
+from rich.console import Console
+
+from sunbeam.clusterd.service import ClusterServiceUnavailableException
+from sunbeam.commands.terraform import TerraformHelper, TerraformInitStep
+from sunbeam.jobs.common import run_plan
+from sunbeam.jobs.juju import JujuHelper
+from sunbeam.plugins.interface.v1.openstack import (
+    DisableOpenStackApplicationStep,
+    EnableOpenStackApplicationStep,
+    OpenStackControlPlanePlugin,
+    TerraformPlanLocation,
+)
+
+LOG = logging.getLogger(__name__)
+console = Console()
+
+
+class ValidationPlugin(OpenStackControlPlanePlugin):
+    """Deploy tempest to openstack model."""
+
+    version = Version("0.0.1")
+
+    def __init__(self) -> None:
+        """Initialize the plugin class."""
+        super().__init__(
+            name="validation",
+            tf_plan_location=TerraformPlanLocation.SUNBEAM_TERRAFORM_REPO,
+        )
+
+    def run_enable_plans(self) -> None:
+        """Run plans to enable plugin."""
+        data_location = self.snap.paths.user_data
+        tfhelper = TerraformHelper(
+            path=self.snap.paths.user_common / "etc" / f"deploy-{self.tfplan}",
+            plan=self._get_plan_name(),
+            backend="http",
+            data_location=data_location,
+        )
+        jhelper = JujuHelper(data_location)
+        plan = [
+            TerraformInitStep(tfhelper),
+            EnableOpenStackApplicationStep(tfhelper, jhelper, self),
+        ]
+
+        run_plan(plan, console)
+        click.echo(f"{self.name} plugin enabled.")
+
+    def run_disable_plans(self) -> None:
+        """Run plans to disable the plugin."""
+        data_location = self.snap.paths.user_data
+        tfhelper = TerraformHelper(
+            path=self.snap.paths.user_common / "etc" / f"deploy-{self.tfplan}",
+            plan=self._get_plan_name(),
+            backend="http",
+            data_location=data_location,
+        )
+        jhelper = JujuHelper(data_location)
+        plan = [
+            TerraformInitStep(tfhelper),
+            DisableOpenStackApplicationStep(tfhelper, jhelper, self),
+        ]
+
+        run_plan(plan, console)
+        click.echo(f"{self.name} plugin disabled.")
+
+    def set_application_names(self) -> list:
+        """Application names handled by the terraform plan."""
+        return ["tempest"]
+
+    def set_tfvars_on_enable(self) -> dict:
+        """Set terraform variables to enable the application."""
+        return {
+            "enable-validation": True,
+            "tempest-channel": "latest/edge",
```

Comment:
not-blocking: could we convert this to constant

Reply:
Sure